GameServerLib/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -588,159 +588,167 @@ public void AddBuff(IBuff b)
+                            // Increment the number of stacks on the parent buff, which is the buff instance which is used for packets.
+                            ParentBuffs[b.Name].IncrementStackCount();
 
-                        BuffSlots[oldestbuff.Slot] = tempbuffs[0];
-                        ParentBuffs.Add(oldestbuff.Name, tempbuffs[0]);
-                        BuffList.Add(b);
+                            // Increment the number of stacks on every buff of the same name (so if any of them become the parent, there is no problem).
+                            GetBuffsWithName(b.Name).ForEach(buff => buff.SetStacks(ParentBuffs[b.Name].StackCount));
 
-                        if (!b.IsHidden)
-                        {
-                            // If the buff is a counter buff (ex: Nasus Q stacks), then use a packet specialized for big buff stack counts (int.MaxValue).
-                            if (ParentBuffs[b.Name].BuffType == BuffType.COUNTER)
-                            {
-                                _game.PacketNotifier.NotifyNPC_BuffUpdateNumCounter(ParentBuffs[b.Name]);
-                            }
-                            // Otherwise, use the normal buff stack (254) update (usually just adds one to the number on the icon and refreshes the time of the icon).
-                            else
+                            if (!b.IsHidden)
                             {
-                                _game.PacketNotifier.NotifyNPC_BuffUpdateCount(b, b.Duration, b.TimeElapsed);
+                                if (b.BuffType == BuffType.COUNTER)
+                                {
+                                    _game.PacketNotifier.NotifyNPC_BuffUpdateNumCounter(ParentBuffs[b.Name]);
+                                }
+                                else
+                                {
+                                    _game.PacketNotifier.NotifyNPC_BuffUpdateCount(b, b.Duration, b.TimeElapsed);
+                                }
                             }
-                        }
-                        b.ActivateBuff();
+                            b.ActivateBuff();
+                            break;
 
-                        return;
-                    }
-                    // If we haven't hit the max stack count (usually 254).
-                    BuffList.Add(b);
-
-                    // Increment the number of stacks on the parent buff, which is the buff instance which is used for packets.
-                    ParentBuffs[b.Name].IncrementStackCount();
+                        // If the buff is supposed to add a stack to any existing buffs of the same name and refresh their timer.
+                        // Essentially the method is: have one parent buff which has the stacks, and just refresh its time, this means no overlapping buff instances, but functionally it is the same.
+                        case BuffAddType.STACKS_AND_RENEWS:
```

Comment:
In the original code it checked `ParentBuffs[b.Name].BuffAddType == BuffAddType.STACKS_AND_RENEWS` while here you check `b.BuffAddType`. I'm not sure what exactly it is changing, but the code is not doing exactly the same thing.

Reply:
An overlook by me.
I'll check that when i get back home

Reply:
From my testing, everything is working as it is supposed to.
ParentBuffs[b.Name] seems to be calling an Ibuff using a "b.name" string key, which will in turn just return "b" again.
Idk if I'm missing something, but it seems it was just redundant code.